### PR TITLE
Expand Android tests

### LINKED ISSUE #1095: [Android] Add missing blockchain test
Android doesn't have address / signer tests for some blockchain (comparing to Swift)

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Expand Android tests, to catch up with iOS tests.  Part of #1095 .

## Testing instructions

`./tools/android-test`

## Types of changes

* New tests

* New tests

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Review by vikmeup — APPROVED** [review]

**Comment by hewigovens** [comment]
did you sort the `switch` cases manually? easy to forget to do that

**Comment by catenocrypt** [comment]
> did you sort the `switch` cases manually? easy to forget to do that

First it seemed to me that some cases are missing in Android (wrongly), so to find them I started to sort them... eneded up with two identical sorted lists; committed nonetheless.

**Comment by catenocrypt** [comment]
More to go, but this batch is ready for merge.

**Review by hewigovens — APPROVED** [review]

File: android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/CoinAddressDerivationTests.kt [hunk 1]
```diff
@@ -29,66 +29,66 @@ class CoinAddressDerivationTests {
     }
 
     private fun runDerivationChecks(coin: CoinType, address: String?) = when (coin) {
+        AETERNITY -> assertEquals("ak_QDHJSfvHG9sDHBobaWt2TAGhuhipYjEqZEH34bWugpJfJc3GN", address)
```
[the remark this answers is not in the document]
Comment (catenocrypt) [reply android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/CoinAddressDerivationTests.kt]: OK, reverted.  It was only reordering.

**Pull request merged**